datalad_fuse/tests/test_fuse.py

```diff
@@ -1,31 +1,60 @@
+from contextlib import contextmanager
 import os.path
+from pathlib import Path
 import subprocess
+from typing import Iterator, Union
 
 from datalad.api import Dataset
 import pytest
 
 pytestmark = pytest.mark.libfuse
 
 
-@pytest.mark.parametrize("transparent", [False, True])
-def test_fuse(tmp_path, transparent, url_dataset):
-    ds, data_files = url_dataset
+@contextmanager
+def fusing(
+    source_dir: Union[str, Path],
+    mount_dir: Path,
+    transparent: bool = False,
+    wait: bool = False,
+) -> Iterator[Path]:
+    mount_dir.mkdir(parents=True, exist_ok=True)
     if transparent:
         opts = ["--mode-transparent"]
-        dots = [".datalad", ".git", ".gitattributes"]
     else:
         opts = []
-        dots = [".datalad", ".gitattributes"]
     p = subprocess.Popen(
-        ["datalad", "fusefs", "-d", ds.path, "--foreground", str(tmp_path), *opts]
+        [
+            "datalad",
+            "fusefs",
+            "-d",
+            str(source_dir),
+            "--foreground",
+            *opts,
+            str(mount_dir),
+        ]
     )
     # Check that the command didn't fail immediately:
     with pytest.raises(subprocess.TimeoutExpired):
         p.wait(timeout=3)
-    assert sorted(q.name for q in tmp_path.iterdir()) == dots + sorted(data_files)
-    for fname, blob in data_files.items():
-        assert (tmp_path / fname).read_bytes() == blob
-    p.terminate()
+    try:
+        yield mount_dir
```

Comment:
Looking at all `dots` which are tightly linked to `transparent`: May be here to return some data struct with `.mount`, `.dots` etc to be used in the tests?  could later even then provide some `.pid` if some test would desire to interrogate it more or smth like that.

Reply:
oh, nevermind -- `fusing` is not the one responsible for creating the dataset, my bad